File: app/src/main/java/com/example/mytemplate/base/BaseActionApi.java
package com.example.mytemplate.base;

import android.content.Context;

import com.example.mytemplate.utils.UserMessage;

import retrofit2.Response;

public class BaseActionApi {

    private Context context;

    public BaseActionApi(Context context){
        this.context = context;
    }

    protected boolean checkResponseIsSuccess(Response response){
        if(response.code() == 200) return true;

        if(response.code() == 401 && context != null)
            UserMessage.defaultMessage(context, "Problem with authentication!");

        if(response.code() >= 500 && context != null)
            UserMessage.defaultMessage(context, "Something went wrong in server!");

        if(response.body() == null && context != null)
            UserMessage.defaultMessage(context, "Response is empty");

        return false;
    }
}

File: app/src/main/java/com/example/mytemplate/base/BaseActionApi.kt
package com.example.mytemplate.base

import android.content.Context
import com.example.mytemplate.utils.UserMessage
import retrofit2.Response

open class BaseActionApi(private val context: Context?) {

    protected fun checkResponseIsSuccess(response: Response<*>): Boolean = when{
        response.code() == 401 -> {
            UserMessage(context).defaultMessage("Problem with authentication!")
            false
        }

        response.code() >= 500 -> {
            UserMessage(context).defaultMessage("Something went wrong in server!")
            false
        }

        response.body() == null -> {
            UserMessage(context).defaultMessage("Response is empty")
            false
        }

        else -> true
    }
}